test-d/index.test-d.ts

```diff
@@ -0,0 +1,69 @@
+import { expectType, expectAssignable, expectError } from 'tsd';
+
+import { KEYS, getKeys, unionWith, VisitorKeysWritable, VisitorKeys } from "../lib/index.js";
```

Comment:
Is this testing `dist/*.d.ts` files or source files?

I tried to make tests fail by editing `dist/index.d.ts` (I changed the return type of `getKeys()` to `number[]`), but `npm run tsd` didn't complain, so it was probably using the source files directly.

However, if I change this import path to `../`:

```suggestion
import { KEYS, getKeys, unionWith, VisitorKeysWritable, VisitorKeys } from "../";
```

then, it seems to be using `dist/*.d.ts` files, because `npm run tsd` now fails with `33:30 Argument of type readonly number[] is not assignable to parameter of type readonly string[].`

Would it make sense to change this path so that the tests use generated definitions, or it doesn't matter?